gnat2goto/driver/tree_walk.adb

```diff
@@ -1206,12 +1209,24 @@ package body Tree_Walk is
             when N_In => Create_Dummy_Irep,
             when N_Real_Literal => Do_Real_Constant (N),
             when N_If_Expression => Do_If_Expression (N),
-            when N_And_Then => Create_Dummy_Irep,
+            when N_And_Then => Do_And_Then (N),
             when N_Or_Else => Create_Dummy_Irep,
             when N_Qualified_Expression => Create_Dummy_Irep,
             when others                 => raise Program_Error);
    end Do_Expression;
 
+   function Do_And_Then (N : Node_Id) return Irep is
+      L : constant Node_Id := Left_Opnd (N);
+      R : constant Node_Id := Right_Opnd (N);
+      Expr : constant Irep := New_Irep (I_Op_And);
```

Comment:
Check: this should be `&&` (short-circuit boolean) not `&`

Reply:
`I_Op_And` maps to `"and"` in `ireps.adb`, which is the same as `ID_and` in cbmc which is `&&`. `&` is `ID_bitand` in cbmc.